Add select change and input event tests

Slotos opened the pull request:
Chromedriver doesn't emit the input event on selection being made, despite actual Chromium browsers emitting one. There are two open issues on the chromedriver project waiting for attention:

- https://bugs.chromium.org/p/chromedriver/issues/detail?id=4104
- https://bugs.chromium.org/p/chromedriver/issues/detail?id=3940

The added test should put a more formal pressure to implement a fix.

Note: The test is expected to fail with chromedriver.

-----

## Additional information:

### [WHATWG HTML5 living standard](https://html.spec.whatwg.org/multipage/form-elements.html#send-select-update-notifications) states:

> When the user agent is to send select update notifications, queue an element task on the user interaction task source given the select element to run these steps:
>
> 1. Fire an event named input at the select element, with the bubbles and composed attributes initialized to true.
> 1. Fire an event named change at the select element, with the bubbles attribute initialized to true.

### [Webdriver standard working draft](https://www.w3.org/TR/webdriver2/#element-click) states that when clicking option element, webdriver should:

> 1. Let parent node be the element’s container.
> 1. Fire a mouseOver event at parent node.
> 1. Fire a mouseMove event at parent node.
> 1. Fire a mouseDown event at parent node.
> 1. Run the focusing steps on parent node.
> 1. If element is not disabled:
>     1. **Fire an input event at parent node.**
>     1. Let previous selectedness be equal to element selectedness.
>     1. If element’s container has the multiple attribute, toggle the element’s selectedness state by setting it to the opposite value of its current selectedness.
>
>         Otherwise, set the element’s selectedness state to true.
>
>     1. **If previous selectedness is false, fire a change event at parent node.**
> 1. Fire a mouseUp event at parent node.
> 1. Fire a click event at parent node.

Slotos committed:
[webdriver] Add select change and input event tests

Chromedriver doesn't emit the input event on selection being made,
despite actual Chromium browsers emitting one. There are two open
issues on the chromedriver project waiting for attention:

- https://bugs.chromium.org/p/chromedriver/issues/detail?id=4104
- https://bugs.chromium.org/p/chromedriver/issues/detail?id=3940

The added test should put a more formal pressure to implement a fix.

whimboo commented:
Thank you for the test. Would you mind to also update the test `test_option_disabled` in the same file, so that we do not only check that the other option that is disabled is not getting selected, but also `input` event is fired?

Slotos committed:
[webdriver] Add an input event test to a single-enabled-choise select

According to WebDriver W3C working draft, when clicking on an option
element, if the element is not disabled, webdriver should always send
an input event, but only send a select event if the previous
selectedness was false.

See https://www.w3.org/TR/2023/WD-webdriver2-20230807/#element-click
for reference

Slotos commented:
I added an input emission event according to WebDriver W3C draft.

I do believe that this behaviour is not clearly defined in the living standard, however. I used the following html document to test actual browsers' behaviour:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title></title>
  </head>
  <body>
    <select id="select">
      <option value="1" disabled>1</option>
      <option value="2">2</option>
    </select>
    <p>
      Input: 
      <span id="input"></span>
    </p>
    <p>
      Change: 
      <span id="change"></span>
    </p>
    <script type="text/javascript">
    document.querySelector('#select').addEventListener("input", (e) => {
      document.querySelector('#input').textContent = e.target.value;
    })
    document.querySelector('#select').addEventListener("change", (e) => {
      document.querySelector('#change').textContent = e.target.value;
    })
    </script>
  </body>
</html>
```

Picking the only enabled option doesn't fire input nor change events. However, living standard specifies that:

> If the multiple attribute is absent, and the element is not disabled, then the user agent should allow the user to pick an option element in its list of options that is itself not disabled. Upon this option element being _picked_ (either through a click, or through unfocusing the element after changing its value, or through a menu command, or through any other mechanism), and before the relevant user interaction event is queued (e.g. before the click event), the user agent must set the selectedness of the picked option element to true, set its dirtiness to true, and then _send select update notifications_.

Where "picked" is a linked concept that refers solely to the quoted paragraph.

Based on that:

- if we interpret "pick an option element" as "choosing any non-disabled option, regardless of its selectedness", then both input and selected events should be emitted when picking already selected option
- if we interpret "pick an option element" as "choosing a non-disabled option whose selectedness was previously false", then neither input nor change event should be fired

In both cases, behaviour dictated by WedDriver draft is divergent, whereas browsers I tested this with (Firefox, Chrome, Safari) conform to the second interpretation of "pick an option element".
